src/commands/profiles/delete.ts

```diff
@@ -0,0 +1,354 @@
+
+    // define all delete operations
+    const deleteOps: ReadonlyArray<DeleteOp> = [
+      {
+        containerName: "profiles",
+        query: selectFromFiscalCode,
+        queryParameters: paramsFromFiscalCode,
+        relatedOps: [],
+        deleteBlobs: false
+      },
+      {
+        containerName: "messages",
+        query: selectFromFiscalCode,
+        queryParameters: paramsFromFiscalCode,
+        relatedOps: [
+          {
+            containerName: "message-status"
+          }
+        ],
+        deleteBlobs: true
+      },
+      {
+        containerName: "notifications",
+        query: selectFromFiscalCode,
+        queryParameters: paramsFromFiscalCode,
+        queryOptions: { enableCrossPartitionQuery: true },
+        relatedOps: [
+          {
+            containerName: "notification-status"
+          }
+        ],
+        deleteBlobs: false
+      },
+      {
+        containerName: "sender-services",
+        query: selectFromRecipientFiscalCode,
+        queryParameters: paramsFromRecipientFiscalCode,
+        relatedOps: [],
+        deleteBlobs: false
+      }
+    ];
+
+    // define the delete operation set from given inputs
+    // if all flag is enabled all above defined options will be procecess
+    // otherwhise only operations specified in the given flags
+    const deleteOpsToProcess = flags.all
+      ? deleteOps
+      : deleteOps.filter(x => {
+          switch (x.containerName) {
+            case "messages":
+            case "message-status":
+              return flags.message;
+            case "notifications":
+            case "notification-status":
+              return flags.notification;
+            case "profiles":
+              return flags.profile;
+            case "sender-services":
+              return flags.service;
+          }
+        });
+    if (deleteOpsToProcess.length === 0) {
+      cli.error("please specify at least 1 container");
+    }
+
+    const client = new cosmos.CosmosClient({
+      endpoint: connection.endpoint,
+      auth: { key: connection.key }
+    });
+    const database = client.database("agid-documentdb-test");
+
+    // tslint:disable-next-line: no-let
+    let countDeleteItems = 0;
+    const deleteOperations = (ops: ReadonlyArray<DeleteOp>) => {
+      return ops.reduce((p, file) => {
+        return p.then(async () => {
+          (await this.processDeleteOpt(database, storageConnection, file)).map(
+            d => (countDeleteItems += d)
+          );
+        });
+      }, Promise.resolve()); // initial
+    };
+    await deleteOperations(deleteOpsToProcess);
+    if (countDeleteItems > 0) {
+      cli.log(`${countDeleteItems} total items successfully deleted`);
+    }
+  }
+
+  /**
+   * delete the given items from the given container
+   * @param items the cosmos Items to delete
+   * @param container the continer where items are placed
+   */
+  private async deleteItems(
+    items: ReadonlyArray<cosmos.Item> | undefined,
```

Comment:
do not call this method if `items === undefined`, handle this case at the caller site

```suggestion
    items: ReadonlyArray<cosmos.Item>,
```